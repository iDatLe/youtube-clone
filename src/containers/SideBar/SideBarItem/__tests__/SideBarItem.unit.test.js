import React from 'react';
import { shallow } from 'enzyme';
import { SideBarItem } from '../SideBarItem';

describe('SideBarItem', () => {

    test('Renders SideBarItem', () => {
        const wrapper = shallow( 
            <SideBarItem />
        )
        expect(wrapper).toMatchSnapshot();
    });

    test('Check highlights in SideBar', () => {
        const wrapper = shallow(
            <SideBarItem icon='fire' label='Trending' />
        )
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders highlighted SideBarItem with no navigation', () => {
        const wrapper = shallow(
          <SideBarItem  highlighted icon='fire' label='Trending'/>
        );
        expect(wrapper).toMatchSnapshot();
      });
});
