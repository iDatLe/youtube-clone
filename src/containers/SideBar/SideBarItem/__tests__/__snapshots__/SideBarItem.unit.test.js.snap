// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem Check highlights in SideBar 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem Renders SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem Renders highlighted SideBarItem with no navigation 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
